File: src/main/java/com/procurement/submission/model/ocds/StatusDetails.java
package com.procurement.submission.model.ocds;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;
import com.procurement.submission.exception.EnumException;
import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

public enum StatusDetails {
    INVITED("invited"),
    PENDING("pending"),
    VALID("valid"),
    DISQUALIFIED("disqualified"),
    WITHDRAWN("withdrawn"),
    EMPTY("empty");

    private static final Map<String, StatusDetails> CONSTANTS = new HashMap<String, StatusDetails>();
    private final String value;

    static {
        for (final StatusDetails c : values()) {
            CONSTANTS.put(c.value, c);
        }
    }

    StatusDetails(final String value) {
        this.value = value;
    }

    @Override
    public String toString() {
        return this.value;
    }

    @JsonValue
    public String value() {
        return this.value;
    }

    @JsonCreator
    public static StatusDetails fromValue(final String value) {
        final StatusDetails constant = CONSTANTS.get(value);
        if (constant == null) {
            throw new EnumException(StatusDetails.class.getName(), value, Arrays.toString(values()));
        }
        return constant;
    }
}

File: src/main/kotlin/com/procurement/submission/domain/model/enums/StatusDetails.kt
package com.procurement.submission.domain.model.enums

import com.fasterxml.jackson.annotation.JsonCreator
import com.fasterxml.jackson.annotation.JsonValue

enum class StatusDetails(@JsonValue override val key: String) : EnumElementProvider.Key {
    INVITED("invited"),
    PENDING("pending"),
    VALID("valid"),
    DISQUALIFIED("disqualified"),
    WITHDRAWN("withdrawn"),
    EMPTY("empty"),
    ARCHIVED("archived");

    override fun toString(): String = key

    companion object : EnumElementProvider<StatusDetails>(info = info()) {

        @JvmStatic
        @JsonCreator
        fun creator(name: String) = StatusDetails.orThrow(name)
    }
}